LibvirtOvsFetchInterfaceCommandWrapperTest fix (test fails in mac) - skip it if no interfaces with eth and wl

sureshanaparti opened the pull request:
### Description

This PR fixes LibvirtOvsFetchInterfaceCommandWrapperTest (test fails in mac) - skip it if no interfaces with eth and wl.

<!--- Describe your changes in DETAIL - And how has behaviour functionally changed. -->

<!-- For new features, provide link to FS, dev ML discussion etc. -->
<!-- In case of bug fix, the expected and actual behaviours, steps to reproduce. -->

<!-- When "Fixes: #<id>" is specified, the issue/PR will automatically be closed when this PR gets merged -->
<!-- For addressing multiple issues/PRs, use multiple "Fixes: #<id>" -->
<!-- Fixes: # -->

<!--- ******************************************************************************* -->
<!--- NOTE: AUTOMATION USES THE DESCRIPTIONS TO SET LABELS AND PRODUCE DOCUMENTATION. -->
<!--- PLEASE PUT AN 'X' in only **ONE** box -->
<!--- ******************************************************************************* -->

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

### Feature/Enhancement Scale or Bug Severity

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [x] Major
- [ ] Minor
- [ ] Trivial


### Screenshots (if appropriate):


### How Has This Been Tested?

Ran unit test in local (mac).

**Before change:**

```
[INFO] Running com.cloud.hypervisor.kvm.resource.wrapper.LibvirtOvsFetchInterfaceCommandWrapperTest
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.108 s <<< FAILURE! - in com.cloud.hypervisor.kvm.resource.wrapper.LibvirtOvsFetchInterfaceCommandWrapperTest
[ERROR] testGetInterfaceDetailsValidValid(com.cloud.hypervisor.kvm.resource.wrapper.LibvirtOvsFetchInterfaceCommandWrapperTest)  Time elapsed: 0.096 s  <<< ERROR!
java.lang.NullPointerException

[ERROR] Errors: 
[ERROR]   LibvirtOvsFetchInterfaceCommandWrapperTest.testGetInterfaceDetailsValidValid » NullPointer
[ERROR] Tests run: 455, Failures: 0, Errors: 1, Skipped: 2
```

**After change:**

```
[INFO] Running com.cloud.hypervisor.kvm.resource.wrapper.LibvirtOvsFetchInterfaceCommandWrapperTest
[WARNING] Tests run: 2, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.109 s - in com.cloud.hypervisor.kvm.resource.wrapper.LibvirtOvsFetchInterfaceCommandWrapperTest
```

<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, and the tests you ran to -->

#### How did you try to break this feature and the system with this change?

<!-- see how your change affects other areas of the code, etc. -->


<!-- Please read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/main/CONTRIBUTING.md) document -->

sureshanaparti commented:
@blueorangutan package

blueorangutan commented:
@sureshanaparti a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

shwstppr reviewed (approved):
code lgtm

Do we need this in 4.18 as well?
Maybe we can also add `networkInterface.getName().startsWith("en")` which may allow the test to run on Mac

blueorangutan commented:
Packaging result [SF]: ✔️ el7 ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10339

sureshanaparti commented:
> code lgtm
> 
> Do we need this in 4.18 as well? Maybe we can also add `networkInterface.getName().startsWith("en")` which may allow the test to run on Mac
> 
> Maybe something like `networkInterface.getName().matches("^(eth|en|wl).*")` for cleaner code

I think, need this in 4.18, i'll rebase to it.

I added "en" check earlier and removed as it's Libvirt/KVM test, should be fine for unit test - will add that.

sureshanaparti commented:
@blueorangutan package

blueorangutan commented:
@sureshanaparti a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el7 ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10355
